### ISSUE TITLE
Possible License Violations in this Project

### ISSUE THREAD

**Issue by cyphar** [snes9xgit/snes9x#276]
I'm not a lawyer, but I have been working with free software for quite a long time. I'm not a copyright holder for any of the software that you (appear to) violate the licenses of, but I would recommend discussing the situation with the relevant copyright holders (and perhaps talking to a lawyer that is knowledgeable about software licenses).

From a cursory glance at this repository there are at least three (perhaps five) license violations, along with unclear licensing terms for the entire project. The former is a bigger problem than the latter, so I'll just focus on that.

[Your license is a proprietary software license that forbids commercial use.](https://github.com/snes9xgit/snes9x/blob/7a2ca6dc2a9a5a643ceda2859be3d459aa9fb50f/docs/snes9x-license.txt) This license is therefore simply incompatible with any version of the GNU GPL, and is incompatible with the GNU LGPL in certain cases. In addition, even if your license *was* compatible with a copyleft license, the entire project would have to be distributed under that copyleft license (though of course you can use whatever license you like on your portions of the code).

Here are the portions of code I've found that have license terms which conflict with your project's license:

* Some of `jma/*` is licensed under the GNU General Public License version 2 *only*. As it is included in the compiled versions of the software, this means that the corresponding source for the binaries generated (and any derived works of the library in general) must be available under the GNU General Public License version 2. Your license is incompatible with the GNU General Public License version 2, and thus you are violating the license of `jma/*`.

* Most of `jma/*` is licensed under the GNU Lesser General Public License version 2.1 *only*. This is a bit more confusing, but since this code is compiled (AFAICS) into the same binaries as the rest of `jma/*` then the entirety of `jma/*` should still be under the GNU General Public License version 2 *only* -- since the LGPL allows you to "upgrade" to the GPL in most circumstances.

* All of `gtk/*` appears to be licensed under the GNU Lesser General Public License version 2.1 *only*. The problem with this is the same as the LGPL-licensed parts of `jma/*` (though if it is compiled into the same binaries as the GPL parts of `jma/*` then it too should be considered as (effectively) GPL licensed in the final project binaries.

* `filter/snes_ntsc_*` is licensed under the GNU Lesser General Public License version 2.1 *only*. Same points as `gtk/*` (since the code also appears to come from the same project).

* `filter/xbrz*` is licensed under the GNU General Public license **version 3** *only*. This makes life more complicated because `GPL-2.0-only` is incompatible with `GPL-3.0-only`. However, you can combine `LGPL-2.1-only` with `GPL-3.0-only` (making the combination `GPL-3.0-only`) -- so the only code that is causing a problem is the parts of `jma/*` which are `GPL-2.0-only` licensed.

It should be noted that `filter/xbrz*` has a weird additional permission to use it with "the MAME library (or with modified versions of MAME that use the same license as MAME)". As far as I can tell, `snes9x` is not a "modified version of MAME that use the same license as MAME", so this additional permission does not apply.

---

In short, you appear to be violating at least three different projects' licenses (and have been for at least 8 years), and I would recommend having a discussion with all of your various contributors to come to a conclusion on how to proceed.

I started looking at this after seeing #264. The copyright holders of this project appear to be quite protective of their copyright, so I'm hoping they'll see why this is an issue they need to address. I don't have a dog in this fight (aside from being very supportive of free software projects, and not liking to see licenses being violated).

**Comment by cyphar** [comment]
Also note that these apparent license violations also affect people who distribute the binaries you've built (they would also be liable for the copyright infringement).

**Comment by bearoso** [comment]
As far as I know, everything is either dual licensed or granted an exception to be used with Snes9x. Nach even put jma in himself, and xbrz was an exception. I dual-licensed my contributions under LGPL in the unlikely scenario that all the copyright holders got together to relicense under something less restrictive and I could no longer by contacted. IANAL, but I believe LGPL can be used even in commercial products compiled in as long as the source is made available, so even if they weren’t contributed to Snes9x, including them would be OK.

Those license indicators are basically in there to say “this part is also under this license, so you can take it and use it under those terms.”

**Comment by OV2** [comment]
- JMA has been included in s9x by its creator and granted exception, as stated in jma/license.txt
- the gtk port is dual licensed as stated in gtk/doc/LICENSE
- blarrg's snes_ntsc filters were approved by him. Statically linking LGPL should also not be a problem since we provide the source for users to link against their own version of the library ([LGPL faq](https://www.gnu.org/licenses/gpl-faq.html#LGPLStaticVsDynamic))
- xbrz: the author has granted an [exception for snes9x](http://www.snes9x.com/phpbb3/viewtopic.php?p=47236#p47236). I'll see if I can contact him to add the exception to the source files.

**Comment by cyphar** [comment]
Okay, I stand corrected. Thanks for clarifying.

Alright, so that touches on the latter point of "the license terms are very vague". If it would be possible to include a top-level license file that explains the license terms clearly, that would be greatly appreciated (including the points you just made that you have already contacted the copyright holders -- something that wouldn't be possible to know looking at the repository, and may be a cause for concern for people working on forks of the project).

> * JMA has been included in s9x by its creator and granted exception, as stated in jma/license.txt
> * the gtk port is dual licensed as stated in gtk/doc/LICENSE
> * blarrg's snes_ntsc filters were approved by him.

Views on how licenses should be declared vary, but from discussions with distribution lawyers the best practice is to include it in the file especially if it already has a file-specific license header (because if someone copies the source -- the dual-license terms are lost). Not to mention that arguably (though perhaps not legally) the license terms most applicable to a piece of code are the ones "in closest proximity" to it.

> Statically linking LGPL should also not be a problem since we provide the source for users to link against their own version of the library (LGPL faq)

Yup, you're completely right. Had a bit of a brain-fart there (I was more concerned with the GPLv2/3 issue while writing) and forgot that you guys distribute the source code. As an aside, have you informed people who distribute this project that they need to provide source code for those components (though a top-level license file should resolve that issue).

**Comment by qwertymodo** [comment]
It is my layman understanding that the GPL allows for distribution of source to be separate from the distribution of the binaries, so in most cases, assuming the binaries were compiled from unmodified source code acquired from some canonical source such as a public repository, pointing users to that repository (i.e. here) should be sufficient, so long as you are certain that the source will remain available. That last part is the main reason people will usually re-host the source themselves, in order to maintain that assurance instead of relying on a 3rd party, but in the case of an official, canonical repository, it should be a reasonable expectation that it will remain available.

**Comment by cyphar** [comment]
@qwertymodo The view of the Software Freedom Conservancy is that that's only true for GPLv3 (there is no equivalent for &sect;6d in GPLv2 or LGPLv2.1 -- the right to forward the upstream link for downstream distributors is not present in &sect;6d in LGPLv2.1 or in any section of GPLv2), though generally most people don't actually care that much (maybe if the project was being developed by a multi-million dollar company this would be more of an issue).

**Comment by bearoso** [comment]
> Just want to point out that commercial != proprietary. Red Hat/SUSE/Canonical/etc ship commercial products that uses plenty of GPL software, but since they obey the licenses (and produce the source for the free software components they ship) that isn't an issue. No free software (or open source) license prohibits the selling of software -- which is why I refer to your license as proprietary. 

It’s unintuitive to call snes9x’s license not free or open source, since it’s both of those things. Call it “not OSI-approved” in this case. Things are vague enough without including the OSI and FSF pedantry.

**Comment by cyphar** [comment]
I disagree (the terms "free software" and "open source" have very specific meanings, and are not vague at all unless you try to make them vague), but I'll try to not start a flame-war here.

In short, "Not OSI-Approved" or "Not FSF-Approved" are not correct terms either, because it ignores that whether a license is approved is not relevant to whether the license is legally "free software" or "open source". Even [the Wikipedia article](https://en.wikipedia.org/wiki/Snes9x#License) doesn't call `snes9x` "open source". Because it isn't (according to the only widely-used meaning of that term).

**Comment by cyphar** [comment]
I disagree (the terms "free software" and "open source" have very specific meanings in the free software community, and are not vague at all), but I'll try to not start a flame-war here.

In short, "not OSI-approved" (while still correct) is misleading because the definition of "open source" is "follows the OSI definition" (you could even argue that the definition of "open source" for a license is that it is OSI approved as well -- making "open source, but not OSI-approved" contradictory). Similarly "not FSF-approved" would also be misleading because "free software" means "provides the four freedoms". [The Wikipedia article for this project](https://en.wikipedia.org/wiki/Snes9x#License) doesn't call it "open source" because it isn't (according to the only widely-used meaning of that term).

**Comment by cyphar** [comment]
I disagree (the terms "free software" and "open source" have very specific meanings in the free software community, and are not vague at all), but I'll try to not start a flame-war here.

In short, "not OSI-approved" (while still correct) is misleading because the definition of "open source" is "follows the OSI definition" (you could even argue that the definition of "open source" for a license is that it is OSI approved as well -- making "open source, but not OSI-approved" contradictory). Similarly "not FSF-approved" would also be misleading because "free software" means "provides the four freedoms". [The Wikipedia article for this project](https://en.wikipedia.org/wiki/Snes9x#License) doesn't call it "open source" because it isn't (according to the only widely-used meaning of that term -- as the license doesn't obey any of the relevant definitions for "free software", "open source", or even "allowed in Debian").

Please don't take this the wrong way -- I don't mean to say that your project is bad because of the license (though I may personally not like your license choice). I just don't like seeing the term "open source" misused in this fashion, as it leads to confusion by ordinary people. From what I've seen, this project is very useful to a lot of people, and that's pretty awesome.

**Comment by bearoso** [comment]
I share your views on the license, hence my dual-licensed contributions. But the phrase "open" has been around a lot longer than the OSI, and simply means "viewable," as in "open book." I don't subscribe to the notion that some corporation can come along and change the meaning of a word, no matter how good their intentions are.

**Comment by twinaphex** [comment]
Just to add my two cents in (even if there was an issue with the below) -

All the snes9x libretro cores completely strip out:
* JMA support
* xbrz filter support
* ntsc filter support
* GTK code

Byuu and blargg explicitly gave their code to snes9x and offered license exemptions to snes9x. There are no license issues here.

**Comment by OV2** [comment]
I've updated to the newest xBRZ that includes exception text for snes9x. I've also added a root LICENSE file that includes information about the multi-licensed parts.
@bearoso should I also add the copyright part to the LICENSE file? I wanted to keep it short and concise.

**Comment by bearoso** [comment]
No, leave the copyrights out.

**Comment by twinaphex** [comment]
I think with this you can close this issue. You have created a LICENSE file, you have made it even more crystal clear to everybody what the terms are and what the licensing is. There is no 'potential' license violation, there wasn't in the beginning, there isn't now, so might as well just move along.

**Comment by cyphar** [comment]
Yup, this was resolved with 8a0fbe69377fee1ce93118b1c20b3756400fd8d8. Thanks everyone.
